src/audio/dai.c

```diff
@@ -473,6 +473,14 @@ static int dai_params(struct comp_dev *dev,
 			comp_err(dev, "dai_params(): failed to alloc dma buffer");
 			return -ENOMEM;
 		}
+
+		/*
+		 * dma_buffer frame_fmt should be overwritten with value
+		 * received in sof_ipc_comp_config, because value from params
+		 * represents data format for component connected with dai
+		 */
+		buffer_set_params(dd->dma_buffer, params, true);
+		dd->dma_buffer->stream.frame_fmt = dconfig->frame_fmt;
```

Comment:
Simple oneliner audio_stream_override_frame_fmt() could be helpful (1/ easier to grep, 2/ improves encapsulation).

Reply:
added

Reply:
Another solution would be to create a local copy of params, override frame_fmt and just set with a single audio-stream API call.